### PR TITLE
Readying for apache maven deploy

### PR DESCRIPTION
**Pull request by larkz**
Namespace refactoring for mavening.

### CODE REVIEW

File: mctreesearch4j/build.gradle [hunk 1]
```diff
@@ -37,14 +44,25 @@ publishing {
...
     // and adds them to the JAR itself. One can extend this task, to skip certain
     // files or particular types at will
     from { configurations.compileClasspath.collect { it.isDirectory() ? it : zipTree(it) } }
 }
 
+apply from: "${rootDir}/scripts/publish-root.gradle"
+
+ext {
+    PUBLISH_GROUP_ID = 'ca.aqtech'
+    PUBLISH_VERSION = '1.0-SNAPSHOT'
```
Comment (JunTaoLuo) [inline mctreesearch4j/build.gradle:63]: Just to confirm, this is just the dev version right? What will be the RTM version?

File: mctreesearch4j/src/main/kotlin/mctreesearch4j/AdvancedStatelessSolver.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package mctreesearch4j
+
+import kotlin.math.ln
+import kotlin.math.sqrt
+
+open class AdvancedStatelessSolver<TState, TAction>(
```
Comment (JunTaoLuo) [inline mctreesearch4j/src/main/kotlin/mctreesearch4j/AdvancedStatelessSolver.kt:6]: I don't think we should ship this in the library. I don't think capturing the exploration term is a common enough scenario to be included by default. This is something that should be implemented by the user.

Comment (larkz) [reply mctreesearch4j/src/main/kotlin/mctreesearch4j/AdvancedStatelessSolver.kt:6]: I was actually trying to access the internals of the Kotlin implementation from Scala as originally intended. In general its many times more difficult difficult to alter the solver from outside Kotlin, given all the type conversions and syntax issues.

I think I can take out the exploration term extraction, but keep this `AdvancedStatelessSolver`'s `getOptimalHorizon` method or move it into the original solver?

File: mctreesearch4j/build.gradle (hunk 1 above)
Comment (larkz) [reply mctreesearch4j/build.gradle:63]: I'm just trying to get this working for now, is 1.0.0 good? Should we start with 0.0.1-beta?

Comment (JunTaoLuo) [reply mctreesearch4j/build.gradle:63]: Start with 0.0.1-beta then. Reserve 1.0.0 for something we actually mean to ship as a first version.

**Pull request merged**